### ISSUE TITLE
[Bug]: Fatal Exception: java.lang.NullPointerException when calling com.onesignal.core.internal.backend.impl.ParamsBackendService.fetchParams

### ISSUE THREAD

**Issue by dmytroreutov** [OneSignal/OneSignal-Android-SDK#1835]
### What happened?

I'm observing some crash reports in Firebase Crashlytics with the following stack trace:

```
Fatal Exception: java.lang.NullPointerException:
       at com.onesignal.core.internal.backend.impl.ParamsBackendService.fetchParams(ParamsBackendService.kt:38)
       at com.onesignal.core.internal.backend.impl.ParamsBackendService$fetchParams$1.invokeSuspend(ParamsBackendService.kt:12)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(BlockingCoroutine.java:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(BuildersKt__Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(Builders.kt:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(BuildersKt__Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(Builders.kt:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Steps to reproduce?

```Markdown
I'm not sure on the steps. Just seeing the reports in Crashlytics.
```

### What did you expect to happen?

Crashes should not happen

### OneSignal Android SDK version

com.onesignal:OneSignal:[5.0.0, 5.99.99]

### Android version

13, 12

### Specific Android models

```Markdown
* Samsung Galaxy A31
* Samsung Galaxy A52
* Xiaomi Redmi Note 11
* TBA
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@dmytroreutov thank you for reaching out! We're investigating this. In the meantime, if you have any additional information you can share on the frequency of this crash or any other details provided by Crashlytics, it would be very appreciated!

**Comment by dmytroreutov** [comment]
Hi, on the screenshot below you can see additional information from Crashlytics

<img width="1294" alt="image" src="https://github.com/OneSignal/OneSignal-Android-SDK/assets/56252415/22fe0106-a301-4ed6-95c3-2d2caa3f8bae">

**Comment by emawby** [comment]
Hello has anyone been able to reproduce the issue locally? It looks like it could be happening when the app has been in the background and is then foregrounded?

**Comment by sarbyn** [comment]
+1 (versione 5.0.2, on production we got a lot of crash like this)

**Comment by jfishman1** [comment]
This should be fixed in the latest 5.1.2 https://github.com/OneSignal/OneSignal-Android-SDK/releases

Please let us know if you continue to see this issue and provide the full crash log including affected devices and operating system versions.
